browser/DocumentRenderer.js

```diff
@@ -985,7 +986,12 @@ class DocumentRenderer extends DocumentRendererBase {
         watcherDefinition = watcherDefinition.apply(null, [attributes]);
       }
 
-      componentContext.watcher = this._state.getWatcher(watcherDefinition);
+      if (watcherDefinition) {
+        componentContext.watcher = this._state.getWatcher(watcherDefinition);
+      } else {
+        this._logger.warn(util.format(WARN_MISSING_WATCHER, watcherName));
```

Comment:
Why do we just log and render the page, wouldn't it be better to throw in such case?